homeassistant/components/mazda/sensor.py

```diff
@@ -1,187 +1,176 @@
 """Platform for Mazda sensor integration."""
-from homeassistant.components.sensor import SensorEntity
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import (
+    CONF_UNIT_SYSTEM_IMPERIAL,
     LENGTH_KILOMETERS,
     LENGTH_MILES,
     PERCENTAGE,
     PRESSURE_PSI,
 )
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.util.unit_system import UnitSystem
 
 from . import MazdaEntity
 from .const import DATA_CLIENT, DATA_COORDINATOR, DOMAIN
 
 
-async def async_setup_entry(hass, config_entry, async_add_entities):
+@dataclass
+class MazdaSensorEntityDescription(SensorEntityDescription):
+    """Describes a Mazda sensor entity."""
+
+    # Suffix to be appended to the vehicle name to obtain the sensor name
+    name_suffix: str | None = None
+
+    # Function to determine whether the vehicle supports this sensor, given the coordinator data
+    is_supported: Callable[[dict], bool] = lambda data: True
```

Comment:
Could you replace `data` with `status`? This would reduce the length of all the lambda expressions.

Reply:
I considered doing that, but the problem is that in the future I will be adding some sensor entities that depend on values which are not part of 'status'. For example electric vehicle sensors will depend on the 'isElectric' attribute which is not in the 'status' data.

Reply:
Can you take a quick look at this merged PR #63136 where two separate functions (`remote_function` and `account_function`) are used to be able to use shorter lambda function? It's a button entity but maybe the same could apply?
https://github.com/home-assistant/core/blob/812cb4a9d8d6a42bfaf5722c2461f9b1a73e6970/homeassistant/components/bmw_connected_drive/button.py#L117-L122

Reply:
I gave this a try, but unfortunately it didn't make the lambda functions short enough to fit on one line. I think the current approach is probably the best one.